lib/phobos_db_checkpoint.rb

```diff
@@ -29,39 +29,56 @@ def table_name_prefix
     end
 
     def configure(pool_size: nil)
+      @verbose && puts("PhobosDBCheckpoint#configure - pool_size = #{pool_size}")
       load_db_config(pool_size: pool_size)
       at_exit { PhobosDBCheckpoint.close_db_connection }
-      ActiveRecord::Base.establish_connection(db_config)
+      PhobosDBCheckpoint.establish_db_connection
     end
 
     def env
       ENV['RAILS_ENV'] ||= ENV['RACK_ENV'] ||= 'development'
     end
 
     def load_db_config(pool_size: nil)
+      @verbose && puts("PhobosDBCheckpoint#load_db_config - pool_size = #{pool_size}")
+
       @db_config_path ||= ENV['DB_CONFIG'] || DEFAULT_DB_CONFIG_PATH
+      @verbose && puts("PhobosDBCheckpoint#load_db_config - db_config_path = #{@db_config_path}, env = #{env}")
+
       configs = YAML.load(ERB.new(File.read(File.expand_path(@db_config_path))).result)
       @db_config = configs[env]
 
       if pool_size.nil? && Phobos.config
-        pool_size = Phobos.config.listeners.map { |listener| listener.max_concurrency || 1 }.inject(&:+)
+        pool_size = number_of_concurrent_listeners + DEFAULT_POOL_SIZE
```

Comment:
Maybe it is too much to add 5 to what is needed, perhaps like @kalopilato suggested `+ 1` would be enough, WDYT? It is needed for eg healtchecks